lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -494,8 +498,22 @@ monitorStakePools tr gp nl db@DBLayer{..} = do
     mkLatestGarbageCollectionEpochRef = newIORef minBound
 
     initCursor :: IO [BlockHeader]
-    initCursor = atomically $ readPoolProductionCursor (max 100 k)
+    initCursor = do
+        fromDB <- atomically $ readPoolProductionCursor (max 100 k)
+        pure (genesisShelleyHeader:fromDB)
       where k = fromIntegral $ getQuantity getEpochStability
+            -- We only have stake pools since shelley era.
+            -- This is hardcoded for mainnet and will do nothing for other
+            -- networks.
+            -- TODO: query cardano-node for hard fork block (not yet possible)
```

Comment:
> not **yet** possible

Is there something suggesting that it _will_ be?

Reply:
My intention to file a feature request!

Reply:
Let's open the ticket first, and reference it from the TODO comment.

Reply:
https://jira.iohk.io/browse/CAD-1850